plugins/coverbrowser.koplugin/main.lua

```diff
@@ -663,4 +716,81 @@ function CoverBrowser:setupHistoryDisplayMode(display_mode)
     end
 end
 
+local function _FileManagerCollections_updateItemTable(self)
+    -- 'self' here is the single FileManagerCollections instance
+    -- FileManagerCollections has just created a new instance of Menu as 'coll_menu'
+    -- at each display of Collection/Favorites. Soon after instantiation, this method
+    -- is called. The first time it is called, we replace some methods.
+    local display_mode = self.display_mode
+    local coll_menu = self.coll_menu
+
+    if not coll_menu._coverbrowser_overridden then
+        coll_menu._coverbrowser_overridden = true
+
+        -- In both mosaic and list modes, replace original methods with those from
+        -- our generic CoverMenu
+        local CoverMenu = require("covermenu")
+        coll_menu.updateItems = CoverMenu.updateItems
+        coll_menu.onCloseWidget = CoverMenu.onCloseWidget
+
+        if display_mode == "mosaic_image" or display_mode == "mosaic_text" then -- mosaic mode
+            -- Replace some other original methods with those from our MosaicMenu
+            local MosaicMenu = require("mosaicmenu")
+            coll_menu._recalculateDimen = MosaicMenu._recalculateDimen
+            coll_menu._updateItemsBuildUI = MosaicMenu._updateItemsBuildUI
+            -- Set MosaicMenu behaviour:
+            coll_menu._do_cover_images = display_mode ~= "mosaic_text"
+            -- no need for do_hint_opened with Collections
+
+        elseif display_mode == "list_image_meta" or display_mode == "list_only_meta" or
+            display_mode == "list_image_filename" then -- list modes
+            -- Replace some other original methods with those from our ListMenu
+            local ListMenu = require("listmenu")
+            coll_menu._recalculateDimen = ListMenu._recalculateDimen
+            coll_menu._updateItemsBuildUI = ListMenu._updateItemsBuildUI
+            -- Set ListMenu behaviour:
+            coll_menu._do_cover_images = display_mode ~= "list_only_meta"
+            coll_menu._do_filename_only = display_mode == "list_image_filename"
+            -- no need for do_hint_opened with Collections
+
+        end
+        coll_menu._do_hint_opened = BookInfoManager:getSetting("history_hint_opened")
```

Comment:
I would naively just set `coll_menu._do_hint_opened = true` , or `= false` (as you wrote above _no need for do_hint_opened with Collections_). Bur probably not the setting that is named after / targeted for History.